cmd/action/internal/update/update.go

```diff
@@ -1,123 +1,150 @@
 package update
 
 import (
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"net/http"
 	"os"
 	"path/filepath"
 	"time"
 
-	"github.com/hashicorp/hcl/v2/gohcl"
-	"github.com/hashicorp/hcl/v2/hclparse"
-	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/mitchellh/go-homedir"
+	"github.com/pelletier/go-toml/v2"
 	"golang.org/x/mod/semver"
 )
 
 type (
-	reason string
-	update struct {
-		shouldNotify bool
-		version      string
-		reason       reason
-	}
 	// Store is where the latest release data is stored on CLI host
 	Store struct {
-		Version        string    `hcl:"version"`
-		URL            string    `hcl:"url"`
-		StoreUpdatedAt time.Time `hcl:"updated"`
+		Version   string    `toml:"version"`
+		URL       string    `toml:"url"`
+		CheckedAt time.Time `toml:"checkedat"`
 	}
 
+	// LatestRelease implements the required fields from https://api.github.com/repos/ariga/atlas/releases/latest
 	LatestRelease struct {
 		Version string `json:"tag_name"`
 		URL     string `json:"html_url"`
 	}
 )
 
-const (
-	reasonPollingInterval reason = "not enough time between checks - 24 hours"
-	reasonNoVersionUpdate reason = "no version update"
-	reasonVersionUpdate   reason = "version update"
-)
-
 // CheckForUpdate implements a notification to the user when a later release is available
-// Logic is based on similar features in other open source CLIs (For example GitHub):
-// 1. Check release file ~/.local/atlas/release.hcl for latest known release and when was it polled
-// 2. Poll GitHub public CLI for latest release if last poll was less than 24h
-// 3. Store a release file in  with the data
-// 4. If current build version, that is not development, is lower than the latest release, notify user
-func CheckForUpdate(version string) {
-	if _, ok := os.LookupEnv("ATLAS_NO_UPDATE_NOTIFIER"); ok {
+// 1. Check release file ~/.local/atlas/release.toml for latest known release and poll time
+// 2. If last poll was more than 24h, poll GitHub public API https://docs.github.com/en/rest/reference/releases#get-the-latest-release
+// 3. Store the latest release metadata
+// 4. If current build Version, that is not development, is lower than the latest release, notify user
+func CheckForUpdate(version string, logF func(i ...interface{})) {
```

Comment:
this signature is a bit unusual for a public method. method should return an err and its caller should decide what to do with the result. Also, this implementation couples the location of the config file, which IMO should be injected to the method. WDYT? 
in addition, the godoc is usually written as a fluent sentence and begins with the method name.
```go
// CheckForUpdate checks if a new version of Atlas CLI exists. To reduce the frequency in which
// this check is made.. yadda yadda.
```

Reply:
For now, I prefer to keep the logic in the package and not expose it outside to the caller, this might change in the near future on other integrations